### PR TITLE
Track time from upload to version creation

### PR DESCRIPTION
**Pull request by kumar303**

### CODE REVIEW

File: apps/versions/tests.py [hunk 1]
```diff
@@ -1237,6 +1238,24 @@ def test_file_not_multi_package(self):
...
+
+            upload_start = calendar.timegm(
+                self.upload.created.utctimetuple())
+            now = calendar.timegm(datetime.now().utctimetuple())
+            rough_delta = (now - upload_start) * 1000
+            actual_delta = mock_timing.call_args[0][1]
+
+            fuzz = 2000  # 2 seconds
+            assert (actual_delta >= (rough_delta - fuzz) and
+                    actual_delta <= (rough_delta + fuzz))
```
Comment (magopian) [inline apps/versions/tests.py]: `assert (rough_delta - fuzz) <= actual_delta <= (rough_delta + fuzz)`?

**Comment by magopian** [comment]
The upload is created even before the validator is called on it, right?

r+wc

**Comment by kumar303** [comment]
Yes, the upload is created right after the [form is processed](https://github.com/mozilla/olympia/blob/master/apps/devhub/views.py#L623) (i.e. the posted file is received). The next step is validation but there is also some task chaining to finally create the version. We already have validation timing so this new metric would allow us to overlay a slightly more complete snapshot of the end to end version creation process.

File: apps/versions/tests.py (hunk 1 above)
Comment (kumar303) [inline apps/versions/tests.py]: I have no idea what that does but if you promise it does the same thing I can add it

Comment (kumar303) [inline apps/versions/tests.py]: actually, I don't think it will add much of an optimization, it's just hard to read (for me)

Comment (magopian) [inline apps/versions/tests.py]: It's not an optimization, just another way to write it, which may be more readable for some. If it's not for you, it's good the way it is!

Comment (kumar303) [inline apps/versions/tests.py]: I know it's a standard JS trick but the operator precedent rules are not obvious which is why I dislike it.

**Pull request merged**